### ISSUE TITLE
not all globstrings are supported in open_files

### ISSUE THREAD

**Issue by ghislainp** [fsspec/filesystem_spec#957]
open_files is insensitive to some globstring:

fspec.open_files("filename*.txt") --> it's working fine.
fspec.open_files("filename?.txt") --> it's not working.
fspec.open_files("filename[0-9].txt") --> it's not working.

I'd expect all these globstrings to be expanded, as documented in the glob method.

The reason seems to be in get_fs_token_paths:

    if isinstance(paths, (list, tuple, set)):
        paths = expand_paths_if_needed(paths, mode, num, fs, name_function)
    else:
        if "w" in mode and expand:
            paths = _expand_paths(paths, name_function, num)
        elif "*" in paths:                                                   <------- HERE. Are more tests needed for "?" and "["  ?
            paths = [f for f in sorted(fs.glob(paths)) if not fs.isdir(f)]
        else:
            paths = [paths]

**Comment by martindurant** [comment]
You are right, that line could probably use a `glob.has_magic` function call checking for all the characters that imply glob expansion should happen.
